File: app/src/main/java/com/iita/akilimo/utils/enums/EnumProduceType.java
package com.iita.akilimo.utils.enums;

public enum EnumProduceType {
    GARI {
        @Override
        public String produce() {
            return "gari";
        }
    },
    FLOUR {
        @Override
        public String produce() {
            return "flour";
        }
    },
    ROOTS {
        @Override
        public String produce() {
            return "roots";
        }
    },
    CHIPS {
        @Override
        public String produce() {
            return "chips";
        }
    };

    public abstract String produce();
}

File: app/src/main/java/com/akilimo/mobile/enums/EnumProduceType.kt
package com.akilimo.mobile.enums


enum class EnumProduceType {
    MAIZE_GRAIN, MAIZE_FRESH_COB,

    SWEET_POTATO_TUBERS, SWEET_POTATO_FLOUR, UNKNOWN;

    fun produce(): String = when (this) {
        MAIZE_GRAIN -> "grain"
        MAIZE_FRESH_COB -> "fresh_cob"
        SWEET_POTATO_TUBERS -> "tubers"
        SWEET_POTATO_FLOUR -> "flour"
        UNKNOWN -> "Unknown"
    }
}
